test/tools/unified-spec-runner/operations.ts

```diff
@@ -206,8 +206,13 @@ operations.set('close', async ({ entities, operation }) => {
     const cursor = entities.getEntity('cursor', operation.object);
     await cursor.close();
   } catch (e) {
-    const changeStream = entities.getEntity('stream', operation.object);
-    await changeStream.close();
+    try {
+      const changeStream = entities.getEntity('stream', operation.object);
+      await changeStream.close();
+    } catch (e) {
+      const client = entities.getEntity('client', operation.object);
+      await client.close();
```

Comment:
```typescript
const entity = ['cursor', 'stream', 'client']
  .map(e => entities.get(e, operation.object, false))
  .find(e => e != null);

if (!entity) {
	throw new Error("attempted to close non-existant entity - good error message");
}
await entity.close();
```

completely optional suggestion - since we know it'll be exactly one type

Reply:
This refactor would necessitate a refactor of the `getEntity` method since it throws whenever the type and object being retrieved do not match. That refactor seems out of scope for this ticket since I am unsure of the implications of changing that behaviour. Note that when a boolean is passed in as the third argument to `getEntity`, it modifies whether or not the runner throws when the entity requested does not exist. It does not, however, change whether or not it throws on type mismatch.

Reply:
I agree, we don't want to make too many changes to shared parts of the test runner. Perhaps we can take this opportunity to at least remove the nesting:

```ts
  try {
    const cursor = entities.getEntity('cursor', operation.object);
    await cursor.close();
    return;
  } catch {
    // ignore
  }

  try {
    const changeStream = entities.getEntity('stream', operation.object);
    await changeStream.close();
    return;
  } catch {
    // ignore
  }

  try {
    const client = entities.getEntity('client', operation.object);
    await client.close();
    return;
  } catch {
    // ignore
  }
```

But as Bailey said, optional suggestion